AsyncDisplayKitTests/ASCollectionViewFlowLayoutInspectorTests.m

```diff
@@ -345,4 +367,50 @@ - (void)testThatItReturnsNoneWhenNoReferenceSizeIsImplemented
   collectionView.asyncDelegate = nil;
 }
 
+- (void)testThatItThrowsIfNodeConstrainedSizeIsImplementedOnDataSourceButNotOnDelegateLayoutInspector
+{
+  UICollectionViewFlowLayout *layout = [[UICollectionViewFlowLayout alloc] init];
+  ASCollectionView *collectionView = [[ASCollectionView alloc] initWithFrame:CGRectZero collectionViewLayout:layout];
+  
+  id dataSourceAndDelegate = [OCMockObject mockForProtocol:@protocol(InspectorTestDataSourceDelegateProtocol)];
+  ASSizeRange constrainedSize = ASSizeRangeMake(CGSizeZero, CGSizeZero);
+  NSIndexPath *indexPath = [NSIndexPath indexPathForItem:0 inSection:0];
+  NSValue *value = [NSValue value:&constrainedSize withObjCType:@encode(ASSizeRange)];
+  [[[dataSourceAndDelegate expect] andReturnValue:value] collectionView:collectionView constrainedSizeForNodeAtIndexPath:indexPath];
+  collectionView.asyncDataSource = dataSourceAndDelegate;
+  
+  id delegate = [InspectorTestDataSourceDelegateWithoutNodeConstrainedSize new];
+  collectionView.asyncDelegate = delegate;
+  
+  ASCollectionViewLayoutInspector *inspector = [[ASCollectionViewLayoutInspector alloc] initWithCollectionView:collectionView];
+  
+  collectionView.layoutInspector = inspector;
+  XCTAssertThrows([inspector collectionView:collectionView constrainedSizeForNodeAtIndexPath:indexPath]);
+  
+  collectionView.asyncDelegate = dataSourceAndDelegate;
+  XCTAssertNoThrow([inspector collectionView:collectionView constrainedSizeForNodeAtIndexPath:indexPath]);
+}
+
+- (void)testThatItThrowsIfNodeConstrainedSizeIsImplementedOnDataSourceButNotOnDelegateFlowLayoutInspector
+{
+  UICollectionViewFlowLayout *layout = [[UICollectionViewFlowLayout alloc] init];
+  ASCollectionView *collectionView = [[ASCollectionView alloc] initWithFrame:CGRectZero collectionViewLayout:layout];
+  
+  id dataSourceAndDelegate = [OCMockObject mockForProtocol:@protocol(InspectorTestDataSourceDelegateProtocol)];
+  ASSizeRange constrainedSize = ASSizeRangeMake(CGSizeZero, CGSizeZero);
+  NSIndexPath *indexPath = [NSIndexPath indexPathForItem:0 inSection:0];
+  NSValue *value = [NSValue value:&constrainedSize withObjCType:@encode(ASSizeRange)];
+  [[[dataSourceAndDelegate expect] andReturnValue:value] collectionView:collectionView constrainedSizeForNodeAtIndexPath:indexPath];
```

Comment:
Use the `-stub` method instead of `-expect` if we're not concerned with how many times it's called